docs/docs/tutorials/simple_hello_world.md

```diff
@@ -0,0 +1,198 @@
+---
+layout: default
+title: Creating a simple hello world with gRPC-Gateway
+parent: Tutorials
+nav_order: 4
+---
+
+### Creating a simple hello world with gRPC-Gateway
+
+### gRPC-Gateway
+
+We all know that gRPC is not a tool for everything. There are cases where we still want to provide a traditional RESTful JSON API. The reasons can range from maintaining backwards-compatibility to supporting programming languages or clients not well supported by gRPC. But coding another API for REST is quite a time consuming and tedious.
+
+So is there any way to code just once, but can provide APIs in both gRPC and REST at the same time?
+
+The answer is Yes.
+
+### Usage
+
+One way to achieve that is to use gRPC gateway. gRPC gateway is a plugin of the protocol buffer compiler. It reads the protobuf service definitions and generates a proxy server, which translates a RESTful HTTP call into gRPC request.
+
+All we need to do is a small amount of configuration
+in the service. Before start coding, we have to install some
+tools.
+
+Use `go get -u` to download the following packages:
+
+```sh
+go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
+go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
+go get -u github.com/golang/protobuf/protoc-gen-go
+```
+
+### Define and Generate proto files
+
+Before we create a gRPC service, we should create a proto file to define what we need, here we create a file named `hello.proto` to show.
+
+```proto
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "io.grpc.examples.helloworld";
+option java_outer_classname = "HelloWorldProto";
+
+package helloworld;
+
+import "google/api/annotations.proto";
+
+// The greeting service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {
+        option (google.api.http) = {
+        post: "/v1/example/echo"
+        body: "*"
+    };
+  }
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+
+Then we should generate gRPC stub via protoc
+
+```sh
+protoc -I/usr/local/include -I. \
+-I$GOPATH/src \
+-I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
+--go_out=Mgoogle/api/annotations.proto=github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis/google/api,plugins=grpc:. \
+helloworld/helloworld.proto
```

Comment:
Where did you get this from? Our docs haven't looked like this for a good while. Maybe we should just refer to `Generating the stubs` here instead of repeating ourselves?